main.tf

```diff
@@ -0,0 +1,234 @@
+}
+
+data "aws_subnet" "selected" {
+  count = var.subnet_ids != null ? 1 : 0
+
+  id = var.subnet_ids[0]
+}
+
+resource "aws_security_group" "default" {
+  #checkov:skip=CKV2_AWS_5: False positive finding, the security group is attached.
+  count = var.subnet_ids != null ? 1 : 0
+
+  name        = local.name
+  description = "Security group for ESC ${local.name}"
+  vpc_id      = data.aws_subnet.selected[0].vpc_id
+  tags        = var.tags
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_vpc_security_group_egress_rule" "default" {
+  for_each = var.subnet_ids != null && length(var.security_group_egress_rules) != 0 ? { for v in var.security_group_egress_rules : v.description => v } : {}
+
+  cidr_ipv4                    = each.value.cidr_ipv4
+  cidr_ipv6                    = each.value.cidr_ipv6
+  description                  = each.value.description
+  from_port                    = each.value.from_port
+  ip_protocol                  = each.value.ip_protocol
+  prefix_list_id               = each.value.prefix_list_id
+  referenced_security_group_id = each.value.referenced_security_group_id
+  security_group_id            = aws_security_group.default[0].id
+  to_port                      = each.value.to_port
+}
+
+resource "aws_ecs_cluster" "default" {
+  count = var.cluster_arn == null ? 1 : 0
+
+  name = "ecs-scheduled-task"
+}
+
+data "aws_region" "current" {}
+
+data "aws_iam_policy_document" "ecs_task" {
+  dynamic "statement" {
+    for_each = local.s3_export_target
+
+    content {
+      effect  = "Allow"
+      actions = ["s3:PutObject*"]
+      resources = [
+        "arn:aws:s3:::${trimprefix(var.config["export-path"], "s3://")}*",
+        "arn:aws:s3:::${trimprefix(var.config["export-path"], "s3://")}"
+      ]
+    }
+  }
+
+  dynamic "statement" {
+    for_each = local.single_account_id
+
+    content {
+      effect = "Allow"
+      actions = [
+        "iam:ListAccountAliases",
+        "ec2:DescribeRegions",
+        "securityhub:ListFindingAggregators",
+        "securityhub:GetFindings",
+        "securityhub:ListEnabledProductsForImport"
+      ]
+      resources = ["*"]
+    }
+  }
+
+  dynamic "statement" {
+    for_each = local.multiple_account_id
+
+    content {
+      effect = "Allow"
+      actions = [
+        "organizations:DescribeOrganization",
+        "organizations:ListAccounts",
+        "organizations:DescribeAccount",
+        "iam:ListAccountAliases",
+        "ec2:DescribeRegions",
+        "securityhub:ListFindingAggregators",
+        "securityhub:GetFindings",
+        "securityhub:ListEnabledProductsForImport"
+      ]
+      resources = ["*"]
+    }
+  }
+}
+
+module "task_role" {
+  source  = "schubergphilis/mcaf-role/aws"
+  version = "~> 0.4.0"
+
+  name                  = local.ecs_task_iam_name
+  create_policy         = true
+  permissions_boundary  = null
```

Comment:
can you create a variable to pass down the permissions_boundary arn?

Reply:
same for the other roles, maybe 1 var to rule them all

Reply:
sure